WalletWasabi.Fluent/ViewModels/Wallets/History/HistoryViewModel.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Reactive.Linq;
+using System.Threading.Tasks;
+using DynamicData;
+using ReactiveUI;
+using WalletWasabi.Blockchain.Transactions;
+using WalletWasabi.Logging;
+using WalletWasabi.Stores;
+using WalletWasabi.Wallets;
+
+namespace WalletWasabi.Fluent.ViewModels.Wallets.History
+{
+	public partial class HistoryViewModel
+	{
+		private readonly Wallet _wallet;
+		private readonly BitcoinStore _bitcoinStore;
+
+		[AutoNotify] private ObservableCollection<HistoryItemViewModel> _histories;
+		[AutoNotify] private bool _showCoinJoin;
+
+		public HistoryViewModel(Wallet wallet, BitcoinStore bitcoinStore)
+		{
+			_wallet = wallet;
+			_bitcoinStore = bitcoinStore;
+			_histories = new ObservableCollection<HistoryItemViewModel>();
+
+			this.WhenAnyValue(x => x.ShowCoinJoin)
+				.ObserveOn(RxApp.MainThreadScheduler)
+				.Subscribe(async _ => await UpdateHistoryAsync());
+		}
+
+		public async Task UpdateHistoryAsync()
```

Comment:
rename to `UpdateAsync` we know its history from the class name.

Reply:
done